### PR TITLE
[go] Add support for version 1.18

### PR DESCRIPTION
**Pull request by sorcererxw**
### Related Issues

Go 1.18 has been released, but I find that I can't deploy my project on Vercel when `go.mod` has `go 1.18` in it.

1.18 adds some very useful features. It would be very valuable for developers to have Vercel support 1.18.

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [ ] ~Issue from task tracker has a link to this PR~

### CODE REVIEW

File: packages/go/go-helpers.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { join } from 'path';
 import stringArgv from 'string-argv';
 import { debug } from '@vercel/build-utils';
 const versionMap = new Map([
+  ['1.18', '1.18.0'],
```
Comment (styfle) [inline packages/go/go-helpers.ts]:
```suggestion
  ['1.18', '1.18.1'],
```

**Review by styfle — APPROVED** [review]
Thanks!

**Comment by sorcererxw** [comment]
@styfle Will Vercel CI support go 1.18 after the next release of this package?

**Comment by styfle** [comment]
Yes in the next stable release. You can try it out today on the canary channel by adding an environment variable to your project: `FORCE_RUNTIME_TAG=canary`

**Pull request merged**